homeassistant/components/ccm15/climate.py

```diff
@@ -0,0 +1,166 @@
+"""Climate device for CCM15 coordinator."""
+import logging
+from typing import Any
+
+from ccm15 import CCM15DeviceState
+
+from homeassistant.components.climate import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    SWING_OFF,
+    SWING_ON,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import CONST_CMD_FAN_MAP, CONST_CMD_STATE_MAP, DOMAIN
+from .coordinator import CCM15Coordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class CCM15Climate(CoordinatorEntity[CCM15Coordinator], ClimateEntity):
+    """Climate device for CCM15 coordinator."""
+
+    def __init__(
+        self, ac_host: str, ac_index: int, coordinator: CCM15Coordinator
+    ) -> None:
+        """Create a climate device managed from a coordinator."""
+        super().__init__(coordinator)
+        self._ac_host: str = ac_host
+        self._ac_index: int = ac_index
+        self._attr_should_poll = True
+        self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        self._attr_has_entity_name = True
+        self._attr_name = f"{self._ac_index}"
+        self._attr_target_temperature_step = 1
+        self._attr_hvac_modes = [
+            HVACMode.OFF,
+            HVACMode.HEAT,
+            HVACMode.COOL,
+            HVACMode.DRY,
+            HVACMode.AUTO,
+        ]
+        self._attr_fan_modes = [FAN_AUTO, FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+        self._attr_swing_modes = [SWING_OFF, SWING_ON]
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.SWING_MODE
+        )
+
+    @property
+    def unique_id(self) -> str:
+        """Return unique id."""
+        return f"{self._ac_host}.{self._ac_index}"
+
+    @property
+    def current_temperature(self) -> int | None:
+        """Return current temperature."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            _LOGGER.debug("temp[%s]=%s", self._ac_index, data.temperature)
+            return data.temperature
+        return None
+
+    @property
+    def target_temperature(self) -> int | None:
+        """Return target temperature."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            _LOGGER.debug("set_temp[%s]=%s", self._ac_index, data.temperature_setpoint)
```

Comment:
Instead of debug loggers everywhere, wouldn't it make more sense to implement the diagnostics platform and just dump the coordinator data in there? (Should be separate PR)

Reply:
Will do, this has been open for months, and I really need this integration to submmit.